crates/sargon/src/signing/collector/signatures_collector.rs

```diff
@@ -0,0 +1,2294 @@
+    }
+
+    fn neglected_factors_due_to_irrelevant(
+        &self,
+        factor_sources_of_kind: &FactorSourcesOfKind,
+    ) -> bool {
+        if self.should_neglect_factors_due_to_irrelevant(factor_sources_of_kind)
+        {
+            info!(
+                "Neglecting all factors of kind: {} since they are all irrelevant (all TX referencing those factors have already failed)",
+                factor_sources_of_kind.kind
+            );
+            self.process_batch_response(SignWithFactorsOutcome::irrelevant(
+                factor_sources_of_kind,
+            ));
+            true
+        } else {
+            false
+        }
+    }
+
+    async fn sign_with_factors_of_kind(
+        &self,
+        factor_sources_of_kind: &FactorSourcesOfKind,
+    ) {
+        info!(
+            "Use(?) #{:?} factors of kind: {:?}",
+            &factor_sources_of_kind.factor_sources().len(),
+            &factor_sources_of_kind.kind
+        );
+
+        let interactor = self
+            .dependencies
+            .interactors
+            .interactor_for(factor_sources_of_kind.kind);
+        let factor_sources = factor_sources_of_kind.factor_sources();
+        match interactor {
+            // PolyFactor Interactor: Many Factor Sources at once
+            SignInteractor::PolyFactor(interactor) => {
+                // Prepare the request for the interactor
+                debug!("Creating poly request for interactor");
+                let request = self
+                    .request_for_parallel_interactor(factor_sources_of_kind);
+                if !request.invalid_transactions_if_neglected.is_empty() {
+                    info!(
+                        "If factors {:?} are neglected, invalid TXs: {:?}",
+                        request.per_factor_source.keys(),
+                        request.invalid_transactions_if_neglected
+                    )
+                }
+                debug!("Dispatching poly request to interactor: {:?}", request);
+                let response = interactor.sign(request).await;
+                debug!("Got response from poly interactor: {:?}", response);
+                self.process_batch_response(response);
+            }
+
+            // MonoFactor Interactor: One Factor Sources at a time
+            // After each factor source we pass the result to the collector
+            // updating its internal state so that we state about being able
+            // to skip the next factor source or not.
+            SignInteractor::MonoFactor(interactor) => {
+                for factor_source in factor_sources {
+                    // Prepare the request for the interactor
+                    debug!("Creating mono request for interactor");
+                    let request = self.request_for_serial_interactor(
+                        &factor_source.factor_source_id().as_hash().unwrap(), // TODO check that
+                    );
```

Comment:
TODO
All factor sources involved in signature collector are supposed to be of Hash type. We make sure of that in the preprocess method. Just want to make sure.

Reply:
see https://github.com/radixdlt/sargon/pull/235#discussion_r1796762949